syntax = "proto3";
package identity_endorsement;

message EndorsementRequest {
    oneof endorsementRequest {
        PkiEndorsementRequest pkiEndorsementRequest = 2;
    }
}

message EndorsementResponse {
    oneof endorsementResponse {
        PkiEndorsementResponse pkiEndorsementResponse = 2;
    }
}

message PkiEndorsementRequest {
    bytes certificateChainPem = 1;
}

message PkiEndorsementResponse {
    EnclaveEndorsement pkiEndorsement = 1;
}

message EnclaveEndorsements {
    EnclaveEndorsement personalPki = 1;
    EnclaveEndorsement dqPki = 2;
}

message EnclaveEndorsement {
    bytes endorsementCertificateDer = 1;
}

message PkiClaim {
    bytes rootCertificateDer = 1;
}